packages/internal-testing-tools/android/src/main/kotlin/com/datadog/reactnative/internaltesting/DdInternalTestingImplementation.kt

```diff
@@ -0,0 +1,240 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.internaltesting
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.context.NetworkInfo
+import com.datadog.android.api.context.TimeInfo
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.internal.net.FirstPartyHostHeaderTypeResolver
+import com.datadog.android.trace.TracingHeaderType
+import com.datadog.reactnative.DatadogSDKWrapperStorage
+import com.facebook.react.bridge.Promise
+import com.google.gson.Gson
+import okhttp3.HttpUrl
+import kotlin.concurrent.thread
+
+/**
+ * The entry point to use Datadog's internal testing feature.
+ */
+class DdInternalTestingImplementation() {
+    private var wrappedCore: StubSDKCore? = null
+
+    /**
+     * Clears all data for all features.
+     */
+    fun clearData(promise: Promise) {
+        wrappedCore?.clearData()
+        promise.resolve(null)
+    }
+
+    /**
+     * Retrieves the list of events for a given feature.
+     */
+    fun getAllEvents(feature: String, promise: Promise) {
+        val events = wrappedCore?.eventsWritten(feature)
+        val eventsJson = Gson().toJson(events)
```

Comment:
very minor: it is a test package, so it doesn't really matter, but since `Gson` instance is thread-safe and stateless, it can be set as property.